pkg/util/resource/view.go

```diff
@@ -20,12 +20,17 @@ type View interface {
 	view()
 }
 
+// ViewWithConfig is a wrapper that provides config to view
+type ViewWithConfig interface {
+	View
+	SecretKeyAllowlist() []string
+}
```

Comment:
My original thought is that we do not even define this interface in this package, because this package is in `util` which should not be aware of secretkey allowlist.

Reply:
If this is the case, where should we put the struct `AppFileWithConfig`? Seems putting it into package `pkg/lib/config/configsource` is not appropriate, since this is not only for descriptors in the configsource. See if you have any idea?

Reply:
Updated!